app/HutchManager/Services/AgentService.cs

```diff
@@ -78,21 +82,20 @@ public async Task<AgentSummary> Get(int agentId)
   /// <param name="agent"></param>
   /// <returns></returns>
   /// <exception cref="KeyNotFoundException"></exception>
-  public async Task<AgentModel> Set(int id, ManageAgent agent)
+  public async Task<AgentSummary> Set(int id, ManageAgent agent)
```

Comment:
I don't think the edit submission should set the secret - i think the act of clicking "regenerate secret" should do it there and then: generate it, hash it, save the hash and show in the ui the raw value.

So that will move from this method.